pkg/network/usm/monitor.go

```diff
@@ -45,10 +45,11 @@ var (
 
 	// knownProtocols maps individual protocol types, to their specification,
 	// for the Monitor to use during its initialisation.
-	knownProtocols = map[protocols.ProtocolType]protocols.ProtocolSpec{
-		protocols.HTTP:  http.Spec,
-		protocols.HTTP2: http2.Spec,
-		protocols.Kafka: kafka.Spec,
+	knownProtocols = map[string]protocols.ProtocolSpec{
+		protocols.HTTP.String():  http.Spec,
+		protocols.HTTP2.String(): http2.Spec,
+		protocols.Kafka.String(): kafka.Spec,
+		javaTLSProgramName:       javaTLSSpec,
```

Comment:
Why going the string way? if we choose to implement it as a protocol, let's add it to the protocol types

Reply:
I think we should rename protocol into EbpfProgram (and remove the split we have today)
java-tls is not a real protocol, thus forcing it to be added as a protocol type is strange
if we merge protocol and EbpfProgram and protocol, and call it EbpfProgram, then it will make sense to add it into a single list, but for sure it shouldn't be "protocol type"

Reply:
Can you please open a Jira ticket for the decision and implementation of this change? (I would say `StatsableEbpfProgram` or `USMEbpfProgram` or something similar would be better if this is the decision)

Reply:
done

Reply:
Update: discussed it together, and decided that actually having a map (and a dedicated key type) does not bring much. We think it shoud be replaced by an array, and the protocols instances themselves should have a Name() method for the rare instances we need it.
This change should be made into a separate PR.